subgraphs/tornado-cash/src/common/getters.ts

```diff
@@ -0,0 +1,414 @@
+import { Address, ethereum, BigInt, dataSource } from "@graphprotocol/graph-ts";
+
+import { fetchTokenSymbol, fetchTokenName, fetchTokenDecimals } from "./tokens";
+import {
+  BIGDECIMAL_ZERO,
+  Network,
+  INT_ZERO,
+  FACTORY_ADDRESS,
+  ProtocolType,
+  SECONDS_PER_DAY,
+  BIGINT_ZERO,
+  SECONDS_PER_HOUR,
+  RewardTokenType,
+  PROTOCOL_NAME,
+  PROTOCOL_SLUG,
+  PROTOCOL_SCHEMA_VERSION,
+  PROTOCOL_SUBGRAPH_VERSION,
+  PROTOCOL_METHODOLOGY_VERSION,
+  ETH_ADDRESS,
+  ETH_NAME,
+  ETH_SYMBOL,
+  ETH_DECIMALS,
+  TORN_ADDRESS,
+  TORN_NAME,
+  TORN_SYMBOL,
+  TORN_DECIMALS,
+  BNB_ADDRESS,
+  BNB_DECIMALS,
+  BNB_NAME,
+  BNB_SYMBOL,
+} from "./constants";
+import { getUsdPricePerToken } from "../prices";
+import { addToArrayAtIndex } from "../common/utils/arrays";
+import { bigIntToBigDecimal } from "./utils/numbers";
+
+import { TornadoCashETH } from "../../generated/TornadoCash01/TornadoCashETH";
+import { TornadoCashERC20 } from "../../generated/TornadoCash01/TornadoCashERC20";
+import { TornadoCashBNB } from "../../generated/TornadoCash01/TornadoCashBNB";
+import {
+  Protocol,
+  Pool,
+  PoolDailySnapshot,
+  PoolHourlySnapshot,
+  Token,
+  RewardToken,
+  UsageMetricsHourlySnapshot,
+  UsageMetricsDailySnapshot,
+  FinancialsDailySnapshot,
+} from "../../generated/schema";
+
+export function getOrCreateProtocol(): Protocol {
+  let network = dataSource.network().toUpperCase();
+  let protocol = Protocol.load(FACTORY_ADDRESS.get(network)!.toHexString());
```

Comment:
See, here instead of using `FACTORY_ADDRESS` directly it would make sense to call `configurations.getFactoryAddress()`. That's what you have the configurations for 😅